openassessment/xblock/static/js/src/lms/oa_base.js

```diff
@@ -68,6 +69,108 @@ OpenAssessment.BaseView.prototype = {
     },
 
     /**
+     * Check if the Aria live region already exists, if it doesn't then add it.
+     * Add the text messages to the Aria live region.
+     *
+     * @param {string[]} texts
+     */
+    srReadTexts: function(texts) {
+        var readerFeedbackID = '.sr.reader-feedback',
+            $readerFeedbackSelector = $(readerFeedbackID),
+            htmlFeedback = '';
+
+        if ($readerFeedbackSelector.length === 0) {
+            $(this.element).append(
+                '<div class="sr reader-feedback" aria-live="polite"></div>'
+            );
+            $readerFeedbackSelector = $(readerFeedbackID);
+        }
+        $.each(texts, function(ids, value) {
+            htmlFeedback = htmlFeedback + '<p>' + value + '</p>\n';
+        });
+        $readerFeedbackSelector.html(htmlFeedback);
+    },
+
+    /**
+     * Checks the rendering status of the views that may require Screen Reader Status updates.
+     *
+     * The only views that should be added here are those that require Screen Reader updates when moving from one
+     * step to another.
+     *
+     * @returns {boolean} true if any step's view is still loading.
+     */
+    areSRStepsLoading: function() {
+        return this.responseView.isRendering ||
+            this.peerView.isRendering ||
+            this.selfView.isRendering ||
+            this.gradeView.isRendering ||
+            this.trainingView.isRendering ||
+            this.staffView.isRendering;
+    },
+
+    /**
+     * Updates text int the Aria live region if all sections are rendered and focuses on the specified ID.
+     *
+     * @param {boolean} announceCurrentStatus - true if the current status should be added to the aria-live region
+     *     even if it is now showing
```

Comment:
Maybe some documentation about the fact that status will be added to the aria-live region regardless of this value, if usageID and focusID are defined and the step is showing.